### PR TITLE
feat: added link component

### PR DESCRIPTION
**Pull request by ayadav-eightfold**
Added Link component

### CODE REVIEW

File: src/components/Navlink/Navlink.types.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    /**
+     * Custom class for the modal
+     * @default StackOrder.row
+     */
+    stackOrder?: StackOrder;
+    /**
+     * Where to open the linked document
+     * @default LinkTarget.self
+     */
+    target?: LinkTarget;
```
Comment (ychhabra-eightfold) [inline src/components/Navlink/Navlink.types.ts:29]: remove this as its alerady part of HTMLAnchorElement

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

File: src/components/Navlink/Navlink.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+export const Navlink: FC<NavlinkProps> = React.forwardRef(
+    (
+        {
+            stackOrder = StackOrder.row,
+            url,
+            classNames,
+            children,
+            target = LinkTarget.self,
+            onClick,
```
Comment (ychhabra-eightfold) [inline src/components/Navlink/Navlink.tsx:16]: remove onClick from here

File: src/components/Navlink/Navlink.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React, { Ref, FC } from 'react';
+
+import { StackOrder, NavlinkProps, LinkTarget } from './Navlink.types';
+import { mergeClasses } from '../../shared/utilities';
+
+import styles from './navlink.module.scss';
+
+export const Navlink: FC<NavlinkProps> = React.forwardRef(
```
Comment (ychhabra-eightfold) [inline src/components/Navlink/Navlink.tsx:8]: Can  we call this Link

File: src/components/Navlink/Navlink.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        </Navlink>
+        <h2>Navlink with default behavior disabled</h2>
+        <Navlink
+            url="https://www.facebook.com"
+            onClick={(e) => e.preventDefault()}
+        >
+            <Icon path={IconName.mdiAccount} />
+            User Profile
+        </Navlink>
+        <h2>Primary Navlink with target='_blank'</h2>
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Navlink/Navlink.stories.tsx:34]: Let's hold on this feedback. The stories files will soon not require a layout, deferring to local MDX implementation using semantic HTML.

**Review by dkilgore-eightfold — APPROVED** [review]

**Pull request closed without merging**